### ISSUE TITLE
Presets not saving on runelite close

### ISSUE THREAD

**Issue by AFKLgnd** [antero111/plugin-presets#51]
I am trying to make multiple presets. I made one as a base preset with all of my plugins unchanged and that preset saves and works fine. Every preset I have made after the first does not save when runelite is closed.

I am trying to save them all to config because I play on multiple computers.

**Comment by antero111** [comment]
Hi, iirc you have all of your presets saved to config (the cloud icon), are you able to reproduce this reliably with some steps I could test? I have seen something like this happen before but haven't got to testing and fixing it yet.

**Comment by AFKLgnd** [comment]
Yes. I only have 1 preset right now when I launch runelite. If I were to make a new preset and close runelite it would not save.

**Comment by antero111** [comment]
> If I were to make a new preset and close runelite it would not save.

Is the made preset saved to config or preset folder? How do you close RuneLite?

Which os you play on and do you use Jagex launcher? Are you logged into RuneLite with Google account?

**Comment by AFKLgnd** [comment]
Made preset is saved to config. I close runelite buy logging out and then
clicking x to close.

I play on Windows 10, do not use the Jagex launcher, and I am logged into
runelite with a gmail account.

**Comment by zeroquinc** [comment]
I have the same issue.

Somehow only my 2 existing presets wont get deleted. But every time I make a 3rd one and close RuneLite and start it again the preset is gone.

My other 2 presets are synced with the option `Config` and have no issues.

Exact steps of what I do:

1. Set up different plugin settings.
2. Press the + button in Plugin Presets.
3. Switch the toggle to `Config` in the top.
4. Press `Add all`
5. Close RuneLite.
6. Open RuneLite and it's gone.

I also tried to reverse step 3 and 4 but it made no difference.
OS is Windows 10 22H2. My colleague on Windows 11 has the exact same problem.

**Comment by AFKLgnd** [comment]
I think there could potentially be conflicts when saving a preset to config with plugins from the plugin hub selected. The first preset I made has them all selected and it saves without an issue, but again every subsequent preset gets deleted.

I tried testing if just selecting 1 plugin to change for a preset would save and it did save successfully to config. The plugin I chose randomly was "xp drops".

The next test I did was if it was having issues saving plugins with custom settings for instance bank tags with bank tabs added and it didn't have issue saving those either.

The next test I did was after a preset had been saved successfully if I could "add all" plugins and on relaunch it would keep the plugins that had been saved successfully before, but not the new ones added.

I then added all plugins and then unchecked all plugins with the (E) next to them and the preset saved successfully.

I then reloaded runelite and tried once again to add the plugins from the plugin hub to the preset and it did not save successfully.

The plugins I currently have downloaded from the hub are:
https://i.imgur.com/go7oAZp.png
https://i.imgur.com/olZNvWw.png

**Comment by antero111** [comment]
Hi, I have been testing this issue for some time now and I haven't really been able to reproduce the problems you guys are having. I think that there is some step that I am missing but here is some notes about my testing: 

Logging in with Google to RuneLite works fine and it saves an instance of your RuneLite settings as it should.  Presets saved to config are stored to each settings.properties configuration file per client and when logged in, saved to RuneLite's "cloud settings file". 
Presets stored locally to the preset folder are shared across every client regardless of the client instance or logged in status. Presets saved to config are shown to that instance that can see it e.g. different open clients have their own "instances" and if you store a preset to one clients config, it won't show in the other client. 

> Presets saved to config are shown to that instance that can see it

This works also when logged in to RuneLite e.g. in home pc and then also logged to work pc. When opening client that is logged in, should load all presets stored in the RuneLites cloud config and made. For example if you made a pvm preset and stored it to the cloud at home, opening RL at work should show the made pvm preset. **If not**, then hitting the refresh presets can fix it in the green + context menu.

One important note to that is that RuneLite saves the config when the client closes. So when creating a preset saved to config at home, the presets only gets synced to the work pc after the client that made the presets gets closed and its config saved to RuneLites cloud. This apply to preset saved to config when not logged in to RL, the config still need to get updated (e.g. client need to be closed) to show the config preset in another client config instance.

Regarding to that testing I made, I haven't been able to accidentally delete any preset even when logging in and out of RuneLite and closing and creating presets with different configurations and plugin hub presets.

**Comment by antero111** [comment]
> Somehow only my 2 existing presets wont get deleted.

Can you elaborate some more on this. How exactly presets won't get deleted?

Regarding to plugin hub plugins: there is a real possibility that some hub plugin causes some problems and if so, its really annoying to start fixing those because there will be next plugin that also break something and then we are in an endless cycle of plugins breaking each other. I'm good when the plugins works as if RuneLite was ran in safemode except with plugin presets enabled.

If you guys are willing to do some more testing, try to click the "refresh presets" when replicating your previous steps to see if it makes any changes. Also it would help a lot if someone could test these with a "fresh install" e.g. with very minimal ammount of hub plugins and updated client. RuneLites safemode won't really be helpful because it disables all hub plugins.

**Comment by zeroquinc** [comment]
My issue got fixed.

The problem is that I didn't understand how it is supposed to work. 

Before I used to include all my plugins to every preset, because I thought I needed to include it for it to sync.
When I realized I used it wrong, I deleted all my presets and only added the plugins that actually made a difference, like FPS Control and GPU.

I got 5 presets now and no issues anymore. maybe this is also the problem of @AFKLgnd.

Another nice thing now is that I don't have to modify every plugin in the other preset when I made a change in my main one. I really  did understand it all wrong lol. Glad it's fixed and thanks for the plugin its working like a charm.
